File: packages/volna-matrix-native/android/src/main/java/social/volna/matrix/VolnaMatrixNativeModule.kt
package social.volna.matrix

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class VolnaMatrixNativeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("VolnaMatrixNative")

    Function("getRuntimeInfo") {
      mapOf(
        "available" to true,
        "implementation" to "matrix-rust-sdk-ffi",
        "bindingVersion" to "26.08.13",
        "platform" to "android",
      )
    }
  }
}

File: packages/volna-matrix-native/ios/VolnaMatrixNativeModule.swift
import ExpoModulesCore
import MatrixSDKFFI

public final class VolnaMatrixNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("VolnaMatrixNative")

    Function("getRuntimeInfo") {
      return [
        "available": true,
        "implementation": "matrix-rust-sdk-ffi",
        "bindingVersion": "26.08.11",
        "platform": "ios",
      ]
    }
  }
}
